### PR TITLE
Update validation.js (and comments)

### PR DESCRIPTION
**Pull request by LDannijs**
#### Summary

Picking up #633 to fix the comments in response to @KrishnaIyer

### CODE REVIEW

**Comment by johanstokking** [comment]
What is the problem that we're trying to solve here?

**Comment by LDannijs** [comment]
@KrishnaIyer asked me to take over that PR and to update the comments so.
Or did you mean the validate itself?

**Comment by LDannijs** [comment]
Whoops that mention was an accident

**Comment by johanstokking** [comment]
What is the problem this pull request is trying to solve?

**Comment by LDannijs** [comment]
Running the validation very often results in an `EAGAIN` error. I do not know the details of it, only that it is a memory issue, but this fixes it by slowing the process down and other fixes (read Jaime's original PR for full details).

**Comment by johanstokking** [comment]
> Running the validation very often results in an `EAGAIN` error. I do not know the details of it, only that it is a memory issue, but this fixes it by slowing the process down and other fixes (read Jaime's original PR for full details).

Do you encounter this yourself? Are there reports from users?

**Comment by LDannijs** [comment]
I encounter it everytime i run it, and users also often do.

**Comment by Jaime-Trinidad** [comment]
Many users seem to be encountering this issue. I believe the following code could provide a temporary solution, though it may slow down the process. It still occasionally encounters an `EAGAIN` error, and I haven't fully grasped the underlying cause yet, especially since it doesn't appear to use all the available memory on my end. Interestingly, this issue has never occurred on my VM running Ubuntu.

Additionally, it's important to note that `error` and `close` are mutually exclusive in this code. The code is designed to promise only one outcome: if an error occurs, it rejects the promise. On a successful close, the promise is resolved. However, if it closes with issues, the promise is also rejected.

**Review by johanstokking — COMMENTED** [review]
If the issue is that too many child processes run at the same time, why not run them sequentially instead? What is the point of parallelism if they're scheduled 1 second apart?

`validatePayloadCodecs()` is awaited. So instead of pushing promises to an array and awaiting them, just await every run.

**Comment by LDannijs** [comment]
So i decided to at least give it a shot and run the feedback through chatgpt and it seems it runs really well now. It's now running through it sequentially and it's not giving me any errors. Hope this is what we're looking for.

@Jaime-Trinidad please take a look as well.

File: bin/validate.js [hunk 1]
```diff
@@ -113,16 +109,38 @@ function validatePayloadCodecs(vendorId, payloadEncoding) {
...
-          `bin/runscript -codec-path "${r.fileName}" -routine ${r.routine} -input '${JSON.stringify(r.input)}'`,
-          (err, stdout, stderr) => {
-            if (err) {
-              reject(err);
-            } else if (stderr) {
-              reject(stderr);
+  for (let index = 0; index < runs.length; index++) {
+    const r = runs[index];
+    await new Promise((resolve, reject) => {
+      setTimeout(async () => {
```
Comment (johanstokking) [inline bin/validate.js:115]: You can remove this

Comment (LDannijs) [reply bin/validate.js]: I've made a change that should hopefully be what your intention is.

**Review by johanstokking — APPROVED** [review]

**Review by Jaime-Trinidad — APPROVED** [review]

**Pull request merged**